pkg/logql/log/fmt.go

```diff
@@ -380,7 +385,12 @@ func (lf *LabelsFormatter) Process(ts int64, l []byte, lbs *LabelsBuilder) ([]by
 	lf.currentLine = l
 	lf.currentTs = ts
 
-	var data interface{}
+	var data = stringMapPool.Get().(map[string]string)
+	clear(data)
```

Comment:
(nit) Would it make a difference if the clear() is done in the defer?

Reply:
Unless I am misunderstanding how some of this works under the hood, it seems like unfortunately the text template execution is using references to the data in the map, so if we clear at the exit of this function rather than when we first grab something from the pool the tests will fail.

I've modified things so that there's a wrapper struct for the pool that calls `clear` for us when we grab something from the pool, just to make things cleaner.